gsy_framework/sim_results/electric_blue/all_results.py

```diff
@@ -25,26 +26,29 @@ def __init__(self):
         self.asset_time_series: Dict[
             int, Dict[int, Dict[DateTime, Dict[str, Dict]]]] = defaultdict(
             lambda: defaultdict(lambda: defaultdict(dict)))
+        self.asset_volume_time_series: Dict[
+            str, Dict[AggregationResolution, AssetVolumeTimeSeries]] = {}
+        self.get_asset_volume_time_series_db = get_asset_volume_time_series_db
         self._total_memory_utilization_kb = 0.0
 
-    def update(self, _area_dict: Dict, core_stats: Dict, current_market_slot: str) -> None:
+    def update(self, area_dict: Dict, core_stats: Dict, current_market_slot: str) -> None:
         """
         Update the forward market aggregated results with bids_offers_trades of current slot.
         """
         if not current_market_slot:
             return
         self._buffer_current_asset_stats()
         self._clear_asset_stats()
-        for area_uuid, area_result in core_stats.items():
-            if "forward_market_stats" not in area_result:
-                return
-
-            current_market_dt = str_to_pendulum_datetime(current_market_slot)
-            for market_type_value, market_stats in area_result["forward_market_stats"].items():
-                market_type = int(market_type_value)
-                self._buffer_bids_offers_trades(area_uuid, market_type, market_stats)
-                current_results = handle_forward_results(current_market_dt, market_stats)
-                self._update_stats_and_time_series(current_results, market_type)
+
+        forward_results = self._get_forward_results(core_stats)
+        if not forward_results:
+            return
+
+        current_market_dt = str_to_pendulum_datetime(current_market_slot)
+        for market_type_value, market_stats in forward_results["forward_market_stats"].items():
+            market_type = int(market_type_value)
```

Comment:
Minor question, how come this is not an int ? IIRC this method is called in the simulation_results service, therefore it receives serialized dicts and not objects, therefore it will probably not contain the enum but the enum value (therefore an int)

Reply:
You're correct, however the problem is that integers can not be used as keys in JSON, so the serializer automatically casts it to a string.

```
In [1]: my_dict = {3: 'GSY'}                                                    

In [2]: import json                                                             

In [3]: json.dumps(my_dict)                                                     
Out[3]: '{"3": "GSY"}'
```

Reply:
Got it. Thanks for the clarification!